src/emulator.ts

```diff
@@ -541,4 +564,61 @@ export class EmacsEmulator implements IEmacsController, vscode.Disposable {
       }
     });
   }
+
+  public saveRegister(arg: string): void {
```

Comment:
It would be great if you write tests for the new commands.
Just leave it if it's hard, as I will make some after merging the PR.

Reply:
Understood. Please have a look at how I've modified EmulatorMap and EmacsEmulator to have a textRegister Map(). If this approach looks good, I'll start writing unit tests